R/oc_forward.R

```diff
@@ -0,0 +1,328 @@
+    }
+
+    # check return
+    return <- match.arg(return)
+
+    # check arguments
+    oc_check_query(
+      placename = placename,
+      key = key,
+      bounds = bounds,
+      countrycode = countrycode,
+      language = language,
+      limit = limit,
+      min_confidence = min_confidence,
+      no_annotations = no_annotations,
+      no_dedupe = no_dedupe,
+      no_record = no_record,
+      abbrv = abbrv,
+      add_request = add_request
+    )
+    # process request
+    oc_process(
+      placename = placename,
+      return = return,
+      key = key,
+      bounds = bounds,
+      countrycode = countrycode,
+      language = language,
+      limit = limit,
+      min_confidence = min_confidence,
+      no_annotations = no_annotations,
+      no_dedupe = no_dedupe,
+      no_record = no_record,
+      abbrv = abbrv,
+      add_request = add_request
+    )
+  }
+
+#' @rdname oc_forward
+#' @param data A data frame
+#' @param bind_cols logical Bind source and results data frame?
+#' @param output A character vector of length one indicating whether only
+#'   latitude, longitude and formatted address (\code{short}) or whether all
+#'   results (\code{all}) should be returned.
+#'
+#' @details
+#' \code{oc_forward_df} also accepts unquoted column names for all arguments
+#' except \code{key}, \code{return}, and \code{no_record}.
+#'
+#' @export
+oc_forward_df <-
+  function(data,
+           placename,
+           bind_cols = TRUE,
+           output = c("short", "all"),
+           key = oc_key(),
+           bounds = NULL,
+           countrycode = NULL,
+           language = NULL,
+           limit = 1L,
+           min_confidence = NULL,
+           no_annotations = TRUE,
+           no_dedupe = FALSE,
+           no_record = FALSE,
+           abbrv = FALSE,
+           ...) {
+
+    placename <- data[[deparse(substitute(placename))]]
+
+    bounds_ <- eval(substitute(alist(bounds)))[[1]]
+    if (is.symbol(bounds_)) {
+      bounds_ <- data[[deparse(bounds_)]]
+      if (!is.null(bounds_)) bounds <- bounds_
+    } else if (is.call(bounds_)) {
+      bounds <- eval(bounds_)
+    }
+    if (!is.null(bounds)) bounds <- as.list(bounds)
+
+    countrycode_ <- eval(substitute(alist(countrycode)))[[1]]
+    if (is.symbol(countrycode_)) {
+      countrycode_ <- data[[deparse(countrycode_)]]
+      if (!is.null(countrycode_)) countrycode <- countrycode_
+    } else if (is.call(countrycode_)) {
+      countrycode <- eval(countrycode_)
+    }
+    if (!is.null(countrycode)) countrycode <- as.list(countrycode)
+
+    language_ <- eval(substitute(alist(language)))[[1]]
+    if (is.symbol(language_)) {
+      language_ <- data[[deparse(language_)]]
+      if (!is.null(language_)) language <- language_
+    } else if (is.call(language_)) {
+      language <- eval(language_)
+    }
+    if (!is.null(language)) language <- as.list(language)
+
+    limit_ <- eval(substitute(alist(limit)))[[1]]
```

Comment:
I'm not good at NSE but I wonder whether the code that is so similar for the different arguments could be put in a function?

Reply:
I tried that, but it did not work. The point is that the arguments are not evaluated (yet), which a function does. There probably is a way to do this more succinctly. E.g. with tidyeval. 😉